### ISSUE TITLE
[?] Explicit mixed in totallyTyped mode

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#652]
I'm not sure if this is a bug, or if Psalm is expected to complain on any mixed assignment, even when both LHS and RHS are explicitly typed as mixed. But using pretty much equivalent union type instead of `mixed` gets rid of error, and at the very least emitted issue misses the RHS type:
```php
<?php
class A
{

    /**
     * @var array<string,scalar|array|object|resource|null>
     */
    private $itemsBroadlyTyped = [];

    /**
     * @param scalar|array|object|resource|null $value
     */
    public function setBroadlyTyped(string $key, $value): void
    {
        $this->itemsBroadlyTyped[$key] = $value;
    }

    /**
     * @var array<string,mixed>
     */
    private $itemsMixed = [];

    /**
     * @param mixed $value
     */
    public function setMixed(string $key, $value): void
    {
        $this->itemsMixed[$key] = $value;
    }
}
```
Here, psalm complains on `$this->itemsMixed[$key]` assignment:
```
ERROR: MixedAssignment - q.php:28:9 - Cannot assign  to a mixed type
        $this->itemsMixed[$key] = $value;
```
and the error message fails to specify *what* cannot be assigned to a mixed type (and I wonder why wouldn't it be possible to assign `mixed` to `mixed`).

**Comment by weirdan** [comment]
The message, or the issue emitted?

**Comment by muglug** [comment]
The issue emitted - if you want `mixed`, Psalm shouldn't complain when you get mixed.
